File: ferry-server/src/main/kotlin/cn/luooyii/ferry/server/service/UserService.java
package cn.luooyii.ferry.server.service;

import cn.luooyii.ferry.server.model.UserEntity;

import java.util.List;

public interface UserService {
	public void insert(UserEntity userEntity);

	public void del(UserEntity userEntity);

	/**
	 * 通过登录名得到用户信息
	 * @param loginName
	 * @return
	 */
	public UserEntity getUserEntityByLoginName(String loginName);

	/**
	 * 获取user列表
	 * @param name
	 * @param pageSize
	 * @param start
	 * @return
	 */
	public List<UserEntity> usersList(String name, int pageSize, int start);

	/**
	 * 获取user列表的总量
	 * @param name
	 * @param pageSize
	 * @param start
	 * @return
	 */
	public Integer usersSize(String name, int pageSize, int start);

	/**
	 * 新建用户信息
	 * @param userEntity
	 */
	public void insertUser(UserEntity userEntity);

	/**
	 * 更新用户信息
	 * @param userEntity
	 */
	public void updateUser(UserEntity userEntity);

	/**
	 * 删除用户信息
	 * @param groupId
	 */
	public void deleteUsers(List<String> groupId);

    UserEntity getEntityById(int userid);
}

File: ferry-server/src/main/kotlin/cn/luooyii/ferry/server/service/UserService.kt
package cn.luooyii.ferry.server.service

import cn.luooyii.ferry.server.model.UserEntity

interface UserService {
    fun insert(userEntity: UserEntity)

    fun del(userEntity: UserEntity)

    /**
     * 通过登录名得到用户信息
     * @param loginName
     * @return
     */
    fun getUserEntityByLoginName(loginName: String): UserEntity

    /**
     * 获取user列表
     * @param name
     * @param pageSize
     * @param start
     * @return
     */
    fun usersList(name: String, pageSize: Int, start: Int): List<UserEntity>

    /**
     * 获取user列表的总量
     * @param name
     * @param pageSize
     * @param start
     * @return
     */
    fun usersSize(name: String, pageSize: Int, start: Int): Int?

    /**
     * 新建用户信息
     * @param userEntity
     */
    fun insertUser(userEntity: UserEntity)

    /**
     * 更新用户信息
     * @param userEntity
     */
    fun updateUser(userEntity: UserEntity)

    /**
     * 删除用户信息
     * @param groupId
     */
    fun deleteUsers(groupId: List<String>)

    fun getEntityById(userid: Int): UserEntity
}
